scripts/upgrade.sh

```diff
@@ -113,6 +113,8 @@ function main() {
     download_util_binaries
     get_machine_id
     merge_yaml_specs
+    parse_yaml_into_bash_variables
+    local finalK8sVersion=${KUBERNETES_VERSION}
     apply_bash_flag_overrides "$@"
     parse_yaml_into_bash_variables
```

Comment:
you're calling parse_yaml_into_bash_variables twice and I'd really like to know why

Reply:
The first is before flag overrides are enforced and one after.  We need to parse into variables and extract the "final k8s version" prior to flag overrides because in a 2 minor version upgrade the bash variable of KUBERNETES_VERSION gets overwritten to enforce step upgrades.

Reply:
okay
this really needs a comment then
because I would have read that as a bug